driver-core/src/main/com/mongodb/connection/netty/NettyStream.java

```diff
@@ -152,13 +198,24 @@ public void initChannel(final SocketChannel ch) {
                             enableHostNameVerification(sslParameters);
                         }
                         engine.setSSLParameters(sslParameters);
-                        ch.pipeline().addFirst("ssl", new SslHandler(engine, false));
+                        pipeline.addFirst("ssl", new SslHandler(engine, false));
                     }
+
                     int readTimeout = settings.getReadTimeout(MILLISECONDS);
+                    final long readTimeoutMillis;
                     if (readTimeout > 0) {
```

Comment:
if `NO_SCHEDULE_TIME` was 0 then this could read `if (readTimeout > NO_SCHEDULE_TIME)`